src/main/java/gov/cms/ab2d/contracts/controller/ContractController.java

```diff
@@ -0,0 +1,42 @@
+package gov.cms.ab2d.contracts.controller;
+
+import gov.cms.ab2d.contracts.model.Contract;
+import gov.cms.ab2d.contracts.service.ContractService;
+import java.util.List;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@AllArgsConstructor
+public class ContractController {
+    private final ContractService contractService;
+
+    @GetMapping("/contract/all")
+    public List<Contract> getAllAttestedContracts() {
+        return contractService.getAllContracts();
+    }
+
+    @PostMapping("/contract")
+    public void updateContract(@RequestBody Contract contract) {
+        contractService.updateContract(contract);
+    }
+
+    @GetMapping("/contract")
+    public Contract getContractByValue(@RequestParam("contractId") Optional<Long> contractId, @RequestParam("contractNumber") Optional<String> contractNumber) {
+        if (contractId.isPresent()) {
+            if (contractNumber.isPresent())
+                throw new InvalidContractParamException("Can't have contractId and contractNumber in the same call");
+
+            return contractService.getContractByContractId(contractId.get());
+        } else if (contractNumber.isPresent())
+            return contractService.getContractByContractNumber(contractNumber.get());
+        else {
+            throw new InvalidContractParamException("Must supply contract information");
+        }
+    }
+}
```

Comment:
I would add the endpoint with a path parameter for the contract number "/contract/{contractNo}"

Reply:
It would be odd two have 2 endpoints (contractId and ContractNumber) that are similar use different ways of sending info.

Reply:
One problem I have is if you have two methods with the same path and rest call type. Spring complains

Reply:
{"@timestamp":"2022-12-06T14:30:46.598-08:00","@version":"1","message":"Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'requestMappingHandlerMapping' defined in class path resource [org/springframework/boot/autoconfigure/web/servlet/WebMvcAutoConfiguration$EnableWebMvcConfiguration.class]: Invocation of init method failed; nested exception is java.lang.IllegalStateException: Ambiguous mapping. Cannot map 'contractController' method \ngov.cms.ab2d.contracts.controller.ContractController#getContractByID(Optional)\nto {GET [/contract]}: There is already 'contractController' bean method\ngov.cms.ab2d.contracts.controller.ContractController#getContractByValue(Optional, Optional) mapped.","logger_name":"org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext","thread_name":"main","level":"WARN","level_value":30000}

Reply:
Made the change. Let me know what you think.